packages/manager-react-components/src/hooks/datagrid/useResourcesV6.tsx

```diff
@@ -31,91 +31,86 @@ export function dataType(a) {
   return typeof a;
 }
 
-function sortColumn(type: string, a: any, b: any, desc: boolean) {
+function sortColumn(
+  type: FilterTypeCategories,
+  a: string,
+  b: string,
+  desc: boolean,
+) {
   if (!a || !b) return -1;
   switch (type) {
     case FilterTypeCategories.Numeric:
       return desc
-        ? parseFloat(a) - parseFloat(b)
-        : parseFloat(b) - parseFloat(a);
+        ? parseFloat(b) - parseFloat(a)
+        : parseFloat(a) - parseFloat(b);
     case FilterTypeCategories.Date:
       return desc
-        ? new Date(a).getTime() - new Date(b).getTime()
-        : new Date(b).getTime() - new Date(a).getTime();
+        ? new Date(b).getTime() - new Date(a).getTime()
+        : new Date(a).getTime() - new Date(b).getTime();
     case FilterTypeCategories.Boolean:
-      return desc ? Number(a) - Number(b) : Number(b) - Number(a);
+      return desc ? Number(b) - Number(a) : Number(a) - Number(b);
     case FilterTypeCategories.String:
       return desc
-        ? a
+        ? b
             ?.trim()
             .toLowerCase()
-            ?.localeCompare?.(b?.trim().toLowerCase())
-        : b
+            ?.localeCompare?.(a?.trim().toLowerCase())
+        : a
             .trim()
             ?.toString()
             ?.toLowerCase()
-            ?.localeCompare?.(a?.trim()?.toLowerCase());
+            ?.localeCompare?.(b?.trim()?.toLowerCase());
     default:
       return -1;
   }
 }
 
-function applySearch(items, filters) {
-  if (!filters?.length) return items;
-  return items.filter((item) =>
-    filters.some(
-      ({ key, value }) =>
-        item[key]?.toString()?.toLowerCase().includes(value.toLowerCase()),
-    ),
-  );
-}
-
-/**
- * @deprecated use fetchIcebergV6 from @ovh-ux/manager-core-api
- */
-export const getResourcesV6 = fetchIcebergV6;
-
-export function useResourcesV6<T = unknown>({
+export function useResourcesV6<T extends Record<string, unknown>>({
   route,
   queryKey,
+  queryFn,
+  refetchInterval,
   pageSize = 10,
   columns = [],
-}: IcebergFetchParamsV6 & ResourcesV6Hook) {
+  defaultSorting,
+}: IcebergFetchParamsV6 & ResourcesV6Hook<T>) {
```

Comment:
actually we could updated the type to `Pick<IcebergFetchParamsV6, 'route'> & ResourcesV6Hook<T>` since we only allow to pass the `route` param from `IcebergFetchParamsV6`

Reply:
Acutally, pageSize is also used, but I get your point.
I think we should remove the dependency to `IcebergFetchParamsV6` and update `ResourcesV6Hook` by adding the missing params and renaming it to `ResourcesV6Params`

Reply:
I pushed a new version